extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/indexing/MSQCompactionRunner.java

```diff
@@ -175,6 +187,31 @@ public CompactionConfigValidationResult validateCompactionTask(
                             .orElse(CompactionConfigValidationResult.success());
   }
 
+  private CompactionConfigValidationResult validatePartitionDimensionsNotMultivalued(
+      PartitionsSpec partitionsSpec,
+      DimensionsSpec dimensionsSpec,
+      Set<String> multiValuedDimensions
+  )
+  {
+    List<String> dimensionSchemas = dimensionsSpec.getDimensionNames();
+    if (partitionsSpec instanceof DimensionRangePartitionsSpec
+        && dimensionSchemas != null
+        && multiValuedDimensions != null) {
```

Comment:
Check for `multiValuedDimensions` empty too?

Reply:
Yes makes it a bit more efficient. Thanks